spec/address_book_spec.rb

```diff
@@ -0,0 +1,53 @@
+ require_relative '../models/address_book'
+
+ RSpec.describe AddressBook do  
+   describe "attributes" do
+     it "should respond to entries" do
+       book = AddressBook.new
+       expect(book).to respond_to(:entries)
+     end
+
+     it "should initialize entries as an array" do
+       book = AddressBook.new
+       expect(book.entries).to be_a(Array)
+     end
+
+     it "should initialize entries as empty" do
+       book = AddressBook.new
+       expect(book.entries.size).to eq(0)
+     end
+   end
+
+   context "#add_entry" do
+     it "adds only one entry to the address book" do
+       book = AddressBook.new
+       book.add_entry('Ada Lovelace', '010.012.1815', 'augusta.king@lovelace.com')
+ 
+       expect(book.entries.size).to eq(1)
+     end
+
+    context "#remove_entry" do
+     it "removes only one entry from the address book" do
+       book = AddressBook.new
+       book.add_entry('Ada Lovelace', '010.012.1815', 'augusta.king@lovelace.com')
+       name = "Hillary Feller"
+       phone_number = "201.328.4392"
+       email_address = "hillfeller@gmail.com"
+ 
+       expect(book.entries.size).to eq(2)
+       book.remove_entry(name, phone_number, email_address)
+       expect(book.entries.size).to eq 1
+       expect(book.entries.first.name).to eq("Ada Lovelace")
+     end
+ 
+     it "adds the correct information to entries" do
+       book = AddressBook.new
+       book.add_entry('Ada Lovelace', '010.012.1815', 'augusta.king@lovelace.com')
+       new_entry = book.entries[0]
+ 
+       expect(new_entry.name).to eq('Ada Lovelace')
+       expect(new_entry.phone_number).to eq('010.012.1815')
+       expect(new_entry.email).to eq('augusta.king@lovelace.com')
+     end
+   end
+ end
```

Comment:
if you switch to the more simplified remove method i mentioned in the last assignment then these specs will have to be simplified as well. the remove_entry specs will change to just remove and and only pass the entry not all the variables.